lib/pages/magic_link/join_workspace_magic_link_page.dart

```diff
@@ -0,0 +1,122 @@
+                ],
+              ),
+              (widget.workspaceJoinResponse == null)
+                ? _buildUnAvailableLayout(widget.workspaceJoinResponse)
+                : _buildAvailableLayout(widget.workspaceJoinResponse)
+            ],
+          ),
+        ),
+      ),
+    );
+  }
+
+  Widget _buildAvailableLayout(WorkspaceJoinResponse? workspaceJoinResponse) =>
+    Column(
+      mainAxisAlignment: MainAxisAlignment.center,
+      children: [
+        Image.asset(imageSmileFace, width: 88, height: 88),
+        SizedBox(height: 20),
+        RichText(
+            text: TextSpan(
+              children: <TextSpan>[
+                TextSpan(text: AppLocalizations.of(context)?.joinWorkspaceTitle ?? '',
+                    style: StylesConfig.commonTextStyle.copyWith(fontWeight: FontWeight.w600, fontSize: 20)),
+                TextSpan(text: workspaceJoinResponse?.workspace.name ?? '',
+                    style: StylesConfig.commonTextStyle.copyWith(fontWeight: FontWeight.w600, fontSize: 20, color: Color(0xff004dff))),
+              ],
+            )),
+        SizedBox(height: 36),
+        Text(AppLocalizations.of(context)?.joinWorkspaceSubTitle(workspaceJoinResponse?.workspace.name ?? '') ?? '', style: TextStyle(fontSize: 15), textAlign: TextAlign.center),
+        SizedBox(height: 12),
+        ButtonTextBuilder(Key('button_join_workspace'),
+            onButtonClick: () => _handleClickOnJoinButton())
+            .setHeight(50)
+            .setText(AppLocalizations.of(context)?.joinWorkspaceButton ?? '')
+            .build()
+      ],
+    );
+
+  Widget _buildUnAvailableLayout(WorkspaceJoinResponse? workspaceJoinResponse) => Column(
+    mainAxisAlignment: MainAxisAlignment.center,
+    children: [
+      Image.asset(imageShrugging, width: 98, height: 88),
+      SizedBox(height: 20),
+      Text(AppLocalizations.of(context)?.joinWorkspaceUnAvailableTitle ?? '',
+          style: StylesConfig.commonTextStyle.copyWith(fontWeight: FontWeight.w600, fontSize: 20)),
+      SizedBox(height: 36),
+      Text(AppLocalizations.of(context)?.joinWorkspaceUnAvailableSubTitle ?? '', style: TextStyle(fontSize: 15), textAlign: TextAlign.center),
+      SizedBox(height: 12),
+      ButtonTextBuilder(Key('button_create_company'),
+          onButtonClick: () => _handleClickOnCreateCompanyButton())
+          .setHeight(50)
+          .setText(AppLocalizations.of(context)?.joinWorkspaceCreateCompanyButton ?? '')
+          .build(),
+      SizedBox(height: 14),
+      Text(AppLocalizations.of(context)?.joinWorkspaceUnAvailableNote ?? '', style: TextStyle(fontSize: 13), textAlign: TextAlign.center),
+    ],
+  );
+
+  void _handleClickOnJoinButton() async {
+    await Get.find<AuthenticationCubit>().joinWorkspace(widget.requestedToken, needCheckAuthentication: true);
+    if(!mounted)
+      return;
+    Navigator.of(context).pop();
+  }
+
+  void _handleClickOnCreateCompanyButton() async {
```

Comment:
I do not understand where it is in UI, and why we need that

Reply:
This button is appeared on this screen by the design. When user clicks on the button, it will open console page. Confirmed this in https://github.com/linagora/Twake-Mobile/issues/1022#issuecomment-977673434
![Screen Shot 2021-11-26 at 12 27 06 PM](https://user-images.githubusercontent.com/29337364/143531379-a0319c8d-1432-497f-9820-1c14fbac5b35.png)